suprb/optimizer/solution/ga/mutation.py

```diff
@@ -9,18 +9,37 @@
 
 
 class SolutionMutation(BaseComponent, metaclass=ABCMeta):
-
-    def __init__(self, mutation_rate: float = 0.05):
-        self.mutation_rate = mutation_rate
-
-    def __call__(self, solution: Solution, random_state: RandomState) -> Solution:
+    def __call__(self, solution: Solution, mutation_rate: float, random_state: RandomState) -> Solution:
         pass
 
 
 class BitFlips(SolutionMutation):
     """Flips every bit in the genome with probability `mutation_rate`."""
 
-    def __call__(self, solution: Solution, random_state: RandomState) -> Solution:
-        bit_flips = random_state.random(solution.genome.shape) < self.mutation_rate
+    def __call__(self, solution: Solution, mutation_rate: float, random_state: RandomState) -> Solution:
+        bit_flips = random_state.random(solution.genome.shape) < mutation_rate
+        genome = np.logical_xor(solution.genome, bit_flips)
+        return solution.clone(genome=genome)
+
+    def saga2_mutation(
```

Comment:
I dont really like that this is part of the bitflips class

Reply:
Moved into the saga2 class